core/jvm/src/main/scala/scalaz/zio/RTS.scala

```diff
@@ -36,10 +36,16 @@ trait RTS {
   }
 
   final def unsafeShutdownAndWait(timeout: Duration): Unit = {
+
+    def awaitTermination(es: ExecutorService) = timeout match {
```

Comment:
I think, perhaps, we should add the following to `Duration`:

```scala
def length: Long
def unit: TimeUnit
```

Then you can use these methods here to `awaitTermination` with `timeout.length` and `timeout.unit`. For `Infinity`, choose the largest possible length (`Long.MaxValue`) and `TimeUnit` (`TimeUnit.DAYS`).

Reply:
@jdegoes at some point we've decided that it's irrelevant for the Duration to keep the original TimeUnit. So now we're converting everything to nanos when creating an instance.

Reply:
@jczuchnowski Sure. The only advantage is that you consolidate the logic that converts the `Duration` into something usable by Java.